prboom2/src/gl_texture.c

```diff
@@ -433,40 +436,26 @@ static void gld_AddPatchToTexture_UnTranslated(GLTexture *gltexture, unsigned ch
             return;
           }
 #endif
-          //e6y: Boom's color maps
-          if (gl_boom_colormaps && use_boom_cm && !(comp[comp_skymap] && (gltexture->flags&GLTEXTURE_SKY)))
+          if (gltexture->flags & GLTEXTURE_INDEXED)
           {
-            const lighttable_t *colormap = (fixedcolormap ? fixedcolormap : fullcolormap);
-            if (gltexture->flags & GLTEXTURE_INDEXED)
-            {
-              // [XA] new indexed color mode: store the palette index in
-              // the R channel and get the colormap index in the shader.
-              buffer[pos + 0] = colormap[source[j]];
-              buffer[pos + 1] = 0;
-              buffer[pos + 2] = 0;
-            }
-            else
-            {
-              buffer[pos+0]=playpal[colormap[source[j]]*3+0];
-              buffer[pos+1]=playpal[colormap[source[j]]*3+1];
-              buffer[pos+2]=playpal[colormap[source[j]]*3+2];
-            }
+            // [XA] new indexed color mode: store the palette index in
+            // the R channel and get the colormap index in the shader.
+            buffer[pos + 0] = gltexture->player_cm == INVULN_PLAYER_CM ? colormap[source[j]] : source[j];
```

Comment:
I was wondering why we don't need a comp_sky check here, but there is a separate gld_AddIndexedSkyToTexture covering indexed mode. Just mentioning this for posterity.